### ISSUE TITLE
Fix comparison functions for Enums with multiple possible values

### ISSUE THREAD

**Issue by jerhard** [goblint/analyzer#226]
Currently, the Enums domain is not able to deliver precise information for comparisons, if at least one of the operands has multiple possible values.

See the following failing example. Enums infers the correct set of possible values for x at the assert, but fails at providing the precise value for the comparison itself.

```C
int main(){
    int top = rand();
    int x,y;

    if(top){
        x = 1;
    } else{
        x = 0;
    }

    assert(x<2); // x in {0,1}
    return 0;
}
```

**Comment by jerhard** [comment]
Side Note: For DefExc, the range information could also be used for comparisons (but is currently not).

**Comment by sim642** [comment]
Makes sense to have this because unlike arithmetic binary operators comparisons have only a few fixed possible results which cannot grow the value sets unboundedly.
